docs/raw/serving-community-history.md

```diff
@@ -0,0 +1,315 @@
+- Store nodes have enough storage to persist historical messages for up to 30 days
+- No store nodes have storage to persist historical messages older than 30 days
+- All nodes are honest
+- The network is reliable
+
+Furthermore, it assumes that:
+
+- Community owner nodes have enough storage to persist historical messages older than 30 days
+- Community owner nodes provide archives with historical messages **at least** every 30 days
+- Community owner nodes receive all community messages
+- Community owner nodes are honest
+
+## Overview
+
+The following is a high-level overview of the user flow and features this specification describes.
+
+### Serving community history archives
+
+Community owner nodes go through the following (high level) process to provide community members with message histories (assumes community owner node is available 24/7):
+
+1. Community owner creates a Status community
+2. Community owner enables community history archive support
+3. A special type of channel for distributing magnet links ([Magnet URI scheme](https://en.wikipedia.org/wiki/Magnet_URI_scheme), [Extensions for Peers to Send Metadata Files](https://www.bittorrent.org/beps/bep_0009.html)) is created
+4. Community owner invites members and creates additional channels
+5. Community owner node receives messages and stores them into local database
+6. After 7 days, the community owner node exports and compresses last 7 days worth of messages from database and creates a magnet link from that data via torrent client
+7. Community owner node creates message archive index and bundles the previously generated magnet link into it
+8. Community owner node creates magnet link from index and distributes it to community members via special channel created in step 2) through the Waku network
+9. Every subsequent 7 days, the step 6) - 8) are repeated and bundled with the previously distributed archive
+
+### Serving archives for missed messages
+
+If the community owner node goes offline, it MUST go through the following process:
+
+1. Community owner node restarts
+2. Community owner node requests messages from store nodes for the missed time range
+3. Missed messages are stored into local database
+4. Community owner node creates message archive and magnet link for missed messages
+5. Community oowner node creates new message archive index bundled with newly created magnet link
+6. The new index is is distributed to community members via special channel through the Waku network
+
+### Receiving community history archives
+
+Community member nodes go through the following (high level) process to fetch and restore community message histories:
+
+1. User joins community and becomes community member
+2. By joining a community, member nodes automatically subscribe to special magnet link channel provided by the community
+3. Member node requests message history (last 30 days) of community channels from store nodes
```

Comment:
If the Member node can fetch a magnet link to a binary blob that contains the community's previous history, and the history included in this binary blob starts starts (for example) at 7 days in the past and then goes backwards, then the member node only needs to fetch message history up until the point when the community history archive service takes over, and this might often be less than 30 days

Reply:
The 30 days here are the up to 30 days that store nodes save. When a community member joins a community, that member has 0 days of message history. So what happens is that it will probably fetch last 30 days of message history from store nodes (this is how this works today).

**Then**, now that there's a notion of message archive index signals, member nodes can recognise those and start requesting archives older than 30 days.

Reply:
@PascalPrecht I think what currently what happens is that when a member joins a community, only the messages in the 'message auto-fetch window' are automatically downloaded, I can't remember what this currently defaults to on Desktop but I think it might be 7 days?  If a user wants to fetch messages beyond the 'message auto-fetch window', the user needs to go into the channel they want to fetch messages for an manually trigger a message fetch.  I would love if we could extend this 'message auto-fetch window' to fetch the full 30 days of message history stored on mailservers, however I remember we ran into a bunch of issue when we tried to do this earlier in the year and that's why we landed on a shorter default 'message auto-fetch window'.  

Now for this community history archive service, it's very important that for the Owner Node this 'message auto-fetch window' is lengthened to always try to fetch all messages from a community that have been sent since the Owner Node was last online, even if the Owner Node was (for example) last online 28 days ago.  This should probably be written down in the spec somewhere.

For community members, I would love if the 'message auto-fetch window' default setting could be lengthened to 30 days, however if the issue that prevented us doing this earlier in the year are still present, we might need to settle of a shorter default message auto-fetch window, say 8 days??